packages/peregrine/lib/talons/CartPage/PriceAdjustments/ShippingMethods/useShippingForm.js

```diff
@@ -0,0 +1,161 @@
+import { useCallback } from 'react';
+import { useApolloClient, useMutation, useQuery } from '@apollo/react-hooks';
+
+import { useCartContext } from '../../../../context/cart';
+
+/**
+ * GraphQL currently requires a complete address before it will return
+ * estimated shipping prices, even though it only needs Country, State,
+ * and Zip. Assuming this is a bug or oversight, we're going to mock the
+ * data points we don't want to bother collecting from the Customer at this
+ * step in the process. We need to be very mindful that these values are never
+ * displayed to the user.
+ */
+const MOCKED_ADDRESS = {
+    city: 'city',
+    firstname: 'firstname',
+    lastname: 'lastname',
+    street: ['street'],
+    telephone: 'telephone'
+};
+
+export const useShippingForm = props => {
+    const {
+        getCountriesQuery,
+        getStatesQuery,
+        selectedValues,
+        setShippingMutation,
+        shippingMethodsQuery
+    } = props;
+
+    const [{ cartId }] = useCartContext();
+    const apolloClient = useApolloClient();
+
+    const {
+        data: countriesData,
+        error: countriesError,
+        loading: isCountriesLoading
+    } = useQuery(getCountriesQuery);
+
+    const {
+        data: statesData,
+        error: statesError,
+        loading: isStatesLoading,
+        refetch: fetchStates
+    } = useQuery(getStatesQuery, {
+        variables: { countryCode: selectedValues.country }
+    });
+
+    const [setShippingAddress, { loading: setShippingLoading }] = useMutation(
+        setShippingMutation
+    );
+
+    const handleCountryChange = useCallback(
+        country => {
+            if (country) {
+                fetchStates({
+                    countryCode: country
+                });
+            }
+        },
+        [fetchStates]
+    );
+
+    const handleZipChange = useCallback(
+        zip => {
+            if (zip !== selectedValues.zip) {
```

Comment:
Am I correct in summarizing this logic as "if the zip changes, clear out the available shipping methods"?
Do we want a comment to that effect at the top?

Reply:
Clearly this is self documenting code 📖 comment added in 23e9cb5.